### ISSUE TITLE
Add pre-cache functionality. 

### ISSUE THREAD

**Issue by Helocominc** [vikram25897/flutter_cached_video_player#65]
Adding a preCache to this package will help to avoid caching the whole video. Take for instance social media apps that has a lot of videos to be played and caching all these videos will affect the memory and performance of the app and device. 

preCache will help to just cache only amount of video time like preCaching for 10 secs as video download.

**Comment by jtkeyva** [comment]
+1

**Comment by davthompson** [comment]
+2

**Comment by vikram25897** [comment]
I don't think it would be easy or even possible to stop caching by video duration. But I guess I can limit it by size. For example, limiting each cache file at 10MB or something? Would that be enough?

**Comment by b3nni97** [comment]
I wish that if you have HSL video you could only precache the first part of the video. 
Then you could preload videos like on Instagram or TikTok and when you want to watch them in full, the video is loaded after the precached part.

**Comment by jtkeyva** [comment]
Don't quite know the nuts and bolts of how this could work and after thinking, not sure it's the responsibility of the video player to pre-cache. https://pub.dev/packages/preload_page_view has worked ok for me. Curious what the solution could be.

If you start adding features then the further away it gets from the official video_player package. Hmm
